File: Client/Src/Managers/DataManager.cs
using UnityEngine;

public class DataManager : MonoBehaviour
{
    public SkillTable SkillTable { get; private set; }
    public MonsterTable MonsterTable { get; private set; }
    public ItemTable ItemTable { get; private set; }
    public ShopTable ShopTable { get; private set; }
    public PlayerTable PlayerTable { get; private set; }

    public void Init()
    {
        SkillTable = new SkillTable();
        SkillTable.Load();

        MonsterTable = new MonsterTable();
        MonsterTable.Load();

        ItemTable = new ItemTable();
        ItemTable.Load();

        ShopTable = new ShopTable();
        ShopTable.Load();

        PlayerTable = new PlayerTable();
        PlayerTable.Load();
    }
}

File: Server/src/Table/DataManager.cpp
#include "../../include/Table/DataManager.h"
#include <iostream>

void DataManager::Init(const std::string& dataRoot)
{
    _skillTable.Load(dataRoot + "/SkillTable.json");
    _monsterTable.Load(dataRoot + "/MonsterTable.json");
    _itemTable.Load(dataRoot + "/ItemTable.json");
    _shopTable.Load(dataRoot + "/ShopTable.json");
    _dropTable.Load(dataRoot + "/DropTable.json");
    _playerTable.Load(dataRoot + "/PlayerTable.json");
}
